srml/staking/src/lib.rs

```diff
@@ -1419,16 +1420,16 @@ impl<T: Trait> Module<T> {
 }
 
 impl<T: Trait> session::OnSessionEnding<T::AccountId> for Module<T> {
-	fn on_session_ending(_ending: SessionIndex, start_session: SessionIndex) -> Option<Vec<T::AccountId>> {
-		Self::new_session(start_session - 1).map(|(new, _old)| new)
+	fn on_session_ending(ending: SessionIndex, _: SessionIndex) -> Option<Vec<T::AccountId>> {
+		Self::new_session(ending + 1).map(|(new, _old)| new)
```

Comment:
Shouldn't this actually just be `start_session`?

Reply:
I am not even sure why the function has two arguments: it seems like the latter should always be former + 1?

Reply:
Yeah, if that's an invariant of the session module then we should remove the second parameter.

Reply:
Documentation says:
```
	/// The returned validator set, if any, will not be applied until `next_index`.
	/// `next_index` is guaranteed to be at least `ending_index + 1`, since session indices don't
```

It is at least `ending_index + 1

Reply:
Ok so the invariant is that it increases but no guarantee by how much. So we should just use the second parameter here, without the `- 1` as it was originally (I think that was wrong).

Reply:
Why should we use the second?